### PR TITLE
setup inference-manager-server deployment

### PR DESCRIPTION
**Pull request by guangrui-cloudnatix**

### CODE REVIEW

File: deployments/server/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+metadata:
+  name: {{ include "inference-manager-server.fullname" . }}-grpc
+  labels:
+    {{- include "inference-manager-server.labels" . | nindent 4 }}
+spec:
+  ingressClassName: {{ .Values.global.ingress.ingressClassName }}
+  rules:
+  - http:
+      paths:
+      - path: /v1/chat/completions
```
Comment (kkaneda) [inline deployments/server/templates/ingress.yaml:12]: The gRPC requires a different ingress definition. Let's remove this ingress if we don't need now.

File: deployments/server/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: {{ include "inference-manager-server.fullname" . }}-http
+  labels:
+    {{- include "inference-manager-server.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "inference-manager-server.labels" . | nindent 6 }}
+  endpoints:
+  - port: monitoring
```
Comment (kkaneda) [inline deployments/server/templates/servicemonitor.yaml:12]: Can you add a return code at the end of the file?

File: deployments/server/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```
Comment (kkaneda) [inline deployments/server/templates/servicemonitor.yaml:2]: Let's avoid adding this or this needs to be able to be disable via ahelm value. This doesn't work if `ServiceMonitor` CRD is not defined.

File: deployments/server/values.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# The following default values work if ollama server runs in the same namespace.
+ollamaServerAddr: llm-operator-inference-manager-engine-ollama:8080
+
+replicaCount: 1
+image:
+  repository: public.ecr.aws/v8n3t7y5/llm-operator/inference-manager-server
+  pullPolicy: IfNotPresent
+  version:
+
+ingress:
```
Comment (kkaneda) [inline deployments/server/values.yaml:18]: Remove this?

**Comment by kkaneda** [comment]
Btw, if the plan is to put this in front of `inference-manager-engine`, we need to remove the ingress from `inference-manager-engine`?

**Comment by guangrui-cloudnatix** [comment]
> Btw, if the plan is to put this in front of `inference-manager-engine`, we need to remove the ingress from `inference-manager-engine`?

yes, I think so

File: deployments/server/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          containerPort: {{ .Values.httpPort }}
+          protocol: TCP
+        - name: grpc
+          containerPort: {{ .Values.grpcPort }}
+          protocol: TCP
+        - name: monitoring
+          containerPort: {{ .Values.monitoringPort }}
+          protocol: TCP
+        env:
+        - name: OLLAMA_HOST
```
Comment (kkaneda) [inline deployments/server/templates/deployment.yaml:38]: Do we still need this?

**Pull request merged**